scripts/o3de/tests/test_register.py

```diff
@@ -248,3 +251,158 @@ def find_ancestor_dir(target_file_name: pathlib.PurePath, start_path: pathlib.Pa
             elif expected_manifest_file == pathlib.PurePath('engine.json'):
                 assert gem_path in map(lambda subdir: TestRegisterGem.engine_path / subdir,
                                        self.engine_data.get('external_subdirectories', []))
+
+@pytest.fixture(scope='class')
+def init_register_project_data(request):
+    request.cls.o3de_manifest_data = json.loads(TEST_O3DE_MANIFEST_JSON_PAYLOAD)
+    request.cls.project_data = json.loads(TEST_PROJECT_JSON_PAYLOAD)
+    request.cls.engine_data = json.loads(TEST_ENGINE_JSON_PAYLOAD)
+
+@pytest.mark.usefixtures('init_register_project_data')
+class TestRegisterProject:
+    engine_path = pathlib.PurePath('o3de')
+    project_path = pathlib.PurePath('TestProject')
+
+    @pytest.mark.parametrize(
+        'test_engine_name, engine_version, engine_api_versions,'
+        'registered_gem_versions, project_engine_name, project_engine_version, '
+        'project_gems, project_compatible_engines, project_engine_api_dependencies,'
+        'gem_compatible_engines, gem_engine_api_dependencies,'
+        'force, expected_result', [
+            # passes when registering without version information
+            pytest.param('o3de1', None, None, { 'gem1':'' }, None, None, ['gem1'], None, None, None, None, False, 0),
+            pytest.param('o3de2', '', None, { 'gem1':'' }, None, None, ['gem1'], None, None, None, None, False, 0),
+            # fails when compatible_engines has no match
+            pytest.param('o3de3', '2.3.4', None, { 'gem1':'' }, None, None, ['gem1'], ['o3de3==1.2.3'], None, None, None,  False, 1),
+            pytest.param('o3de4', '1.2.3', None, { 'gem1':'' }, "", "", ['gem1'], ['o3de1'], None, None, None, False, 1),
+            # passes when forced
+            pytest.param('o3de5', '1.2.3', None, { 'gem1':'' }, None, None, ['gem1'], ['o3de1'], None, None, None, True, 0),
+            # passes when compatible_engines has match
+            pytest.param('o3de6', '0.0.0', None, { 'gem1':'' }, None, None, ['gem1'], ['o3de6'], None, None, None, False, 0),
+            pytest.param('o3de7', '1.2.3', None, { 'gem1':'' }, None, None, ['gem1'], ['o3de7>=1.2.3','o3de-sdk==2.3.4'], None, None, None, False, 0),
+            # fails when uses gem that is not known compatible with version 1.2.3 
+            pytest.param('o3de8', '1.2.3', None, { 'gem1':'' }, None, None, ['gem1'], ['o3de8'], None, ['o3de==2.3.4'], None, False, 1),
+            # passes when uses gem that is known compatible with version 1.2.3 
```

Comment:
nitpick: extra word "uses"
```suggestion
            # fails when gem is used that is not known compatible with version 1.2.3 
            pytest.param('o3de8', '1.2.3', None, { 'gem1':'' }, None, None, ['gem1'], ['o3de8'], None, ['o3de==2.3.4'], None, False, 1),
            # passes when gem is used that is known compatible with version 1.2.3 
```